### ISSUE TITLE
Invalid Insufficient funds err on `GasEstimateMessageGas`

### ISSUE THREAD

**Issue by Schwartz10** [filecoin-project/lotus#9035]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [X] lotus client
- [X] lotus JSON-RPC API
- [X] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
This is occurring on the public Glif API nodes running v1.16.0
```

### Describe the Bug

`GasEstimateMessageGas` can return an `exit SysErrInsufficientFunds(6)` error (insufficient funds), but the "funds" amount used is different from the wallet's balance.

This may not actually be a bug, but it _seems_ like an unexpected change in behavior since the last network upgrade incorporating the FVM. 

For instance, let's say I have a wallet with a balance of 1001 attofil, and I attempt to gasEstimateMessageGas with a message that is a send of 1001 attofil from my wallet to another wallet. I wouldn't expect this to return me an insufficient funds error.

However, I do get an insufficient funds error, and the balance that's reported looks like its my balance _minus_ the gas used after this sample message is run through the VM? See curl commands in repro steps 

### Logging Information

```text
N/a but see the repro steps below.
```

### Repo Steps

Sample wallet: `f1yqh7rqkwadnx4foph43c4kfeh3z5ug354qc2umy`

Get the wallet's current balance:

```
curl --location --request POST 'https://api.node.glif.io/rpc/v0' \
--header 'Content-Type: application/json' \
--data-raw '{
"jsonrpc": "2.0",
"method": "Filecoin.WalletBalance",
"id": 1,
"params": ["f1yqh7rqkwadnx4foph43c4kfeh3z5ug354qc2umy"]
}'
```

returns

```
{
    "jsonrpc": "2.0",
    "result": "48841082302129422",
    "id": 1
}
```

So the wallet has a balance of: `48841082302129422` attofil. 

GasEstimateMessageGas of a simple transfer of `48841082302129422` attofil

```
curl --location --request POST 'https://api.node.glif.io/rpc/v0' \
--header 'Content-Type: application/json' \
--data-raw '{
"jsonrpc": "2.0",
"method": "Filecoin.GasEstimateMessageGas",
"id": 1,
"params": [{
	"To": "f1yqh7rqkwadnx4foph43c4kfeh3z5ug354qc2umy",
	"From": "f1yqh7rqkwadnx4foph43c4kfeh3z5ug354qc2umy",
	"Nonce": 0,
	"Value": "48841082302129422",
	"Method": 0,
	"Params": []
}, {"MaxFee": "0"}, null]
}'
```

Returns the error:

```
{
    "jsonrpc": "2.0",
    "id": 1,
    "error": {
        "code": 1,
        "message": "estimating gas used: message execution failed: exit SysErrInsufficientFunds(6), reason: message failed with backtrace:\n--> caused by: send::send -- sender does not have funds to transfer (balance 48840072302129422, transfer 48841082302129422) (5: insufficient funds)\n (RetCode=6)"
    }
}
```

Notice how the balance reported by the error:

`48840072302129422`

Is less than the current balance of the wallet `48841082302129422`

**Comment by arajasek** [comment]
@Schwartz10 This _should_ already be fixed in master (by #8991) -- any chance you can give it a quick try?

**Comment by ArseniiPetrovich** [comment]
@Schwartz10 no, sorry, we only run stable builds. We can run one for you on latest, but on Monday only, sorry :(
